src/qonlinetranslator.cpp

```diff
@@ -2397,14 +2401,15 @@ QString QOnlineTranslator::languageApiCode(Engine engine, Language lang)
 
     switch (engine) {
     case Google:
-    case Lingva: // Lingva is a Google Translate frontend, so will support same langs
         return s_googleLanguageCodes.value(lang, s_genericLanguageCodes.value(lang));
     case Yandex:
         return s_yandexLanguageCodes.value(lang, s_genericLanguageCodes.value(lang));
     case Bing:
         return s_bingLanguageCodes.value(lang, s_genericLanguageCodes.value(lang));
     case LibreTranslate:
         return s_genericLanguageCodes.value(lang);
+    case Lingva:
+        return s_lingvaLanguageCodes.value(lang);
```

Comment:
I think you need to fallback to `s_genericLanguageCodes`:
```suggestion
        return s_lingvaLanguageCodes.value(lang, s_genericLanguageCodes.value(lang)));
```

Reply:
Should I fallback in Libre?

Reply:
Why? You previously used generic language codes.